sds_data_manager/lambda_code/SDSCode/batch_starter.py

```diff
@@ -290,158 +185,188 @@ def is_job_in_status_table(
 
-    # Event details:
-    filename = event["detail"]["object"]["key"]
-    components = ScienceFilePath.extract_filename_components(filename)
-    logger.info(f"Parsed filename - {components}")
-    instrument = components["instrument"]
+    Go through the job information to retrieve all necessary input files
+    (upstream dependencies). If any are missing, return. Otherwise, submit
+    the job to the batch queue.
 
-    # Get information for the batch job.
-    region = os.getenv("REGION")
-    account = os.getenv("ACCOUNT")
-    # Create a batch client
-    batch_client = boto3.client("batch")
+    Parameters
+    ----------
+    session : orm session
+        Database session.
+    batch_client : boto3.client
+        Boto3 client for AWS Batch.
+    job_info : dict
+        Dictionary containing components with dates and versions appended.
+
+    Returns
+    -------
+    bool
+        Whether or not this job is ready to be processed.
+    """
+    instrument = job_info["instrument"]
+    data_level = job_info["data_level"]
+    descriptor = job_info["descriptor"]
+    start_date = job_info["start_date"]
+    version = job_info["version"]
+
+    logger.info("Checking for job in progress before looking for dependencies.")
+
+    if is_job_in_status_table(
+        session=session,
+        instrument=instrument,
+        data_level=data_level,
+        descriptor=descriptor,
+        start_date=start_date,
+        version=version,
+    ):
+        logger.info(
+            f"Job already in progress for {instrument}, {data_level}, "
+            f"{descriptor}, {start_date}, {version}"
+        )
+        return
+
+    # Find the files that this job depends on
+    upstream_dependencies = get_dependencies(
+        session=session,
+        instrument=instrument,
+        data_level=data_level,
+        descriptor=descriptor,
+        direction="UPSTREAM",
+        relationship="HARD",
+    )
+
+    for upstream_dependency in upstream_dependencies:
+        instrument = upstream_dependency["instrument"]
+        data_level = upstream_dependency["data_level"]
+        descriptor = upstream_dependency["descriptor"]
+        # Check to see if each upstream dependency file is available
+        record = get_file(
+            session, instrument, data_level, descriptor, start_date, version
+        )
+        if record is None:
+            logger.info(
+                f"Dependency not found: {instrument}, "
+                f"{data_level}, "
+                f"{descriptor}, "
+                f"{version}"
+            )
+            return False  # Exit the loop early as we already found a missing dependency
+
+        # Add additional information to the upstream dependency
+        # TODO: Should we actually be using the record's information?
+        #      We are currently using our upstream dependency, but it seems like we
+        #      might want to use the file record itself as our source of truth.
```

Comment:
This comment feels outdated but the concept may be same as what I mentioned in previous comment.

Reply:
Yeah, this is definitely confusing right now and it is related to your other comment. I moved this logic up and updated the comments to make it more explicit that we should be changing these in the future. Right now we are using the input-job's start date and version number.